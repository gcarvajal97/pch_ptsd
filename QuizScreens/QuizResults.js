import React, { Component } from "react";
import { View, Text, StyleSheet, } from "react-native";

export default class QuizResults extends Component {

    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
        headerTitle: 'Take the Quiz'
    }

    render() {
        // Receive results using props, apparently only way to do this. RN has an issue with params and StackNavigator at the moment.
        // It still works as intended, but I was trying to use the modern practice of NavigationActions and creating a NavigationService. 
        const results = this.props.navigation.getParam('results', 'nothing found');
        
        return (
            <View style={styles.container}>
                <Text>
                    This is the Quiz Results Screen
                </Text>
                {
                // Just an example of iterating through the results using map and rendering to the page.
                Object.entries(results).map(([key,v])=>{
                    return <Text key={key}>{key +": value = "+ v}</Text>})
                }
        </View>
        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        margin: 10,
        backgroundColor: 'white',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.23,
        shadowRadius: 2.62,

        elevation: 4,
    },
    bullet: {
        marginVertical:1, 
        marginHorizontal:18, 
        textAlign:'left', 
        alignSelf:'stretch', 
        fontWeight:'bold',
        fontSize:14,
        paddingTop: 4,
    },
    paragraph: {
        alignSelf:"center", 
        paddingVertical:1, 
        paddingHorizontal:5,
        fontSize: 14,
        marginTop: 5,
    },
});