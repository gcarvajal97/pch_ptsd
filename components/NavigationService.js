import { NavigationActions } from 'react-navigation';
import { DrawerActions } from 'react-navigation-drawer';

// Props is apparently to be deprecated soon, I've researched the current best practice and created this "service"
// This service will allow us to make calls to move screens using the navigate function below
// I also added in an openDrawer function (so that it can be called from the header button on MainScreen.js)
let _navigator;

function setTopLevelNavigator(navigatorRef) {
  _navigator = navigatorRef;
}

function getTopLevelNavigator() {
  return _navigator;
}

// Can be used to navigate to other screens, but currently Drawer only references the stack navigator
function navigateDrawer(routeName, params) {
  _navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
}

// Can be used to move into other screens defined in the stack navigator. 
function navigate(routeName, params) {
    _navigator.dispatch(
      NavigationActions.navigate({
        routeName,
        params,
      })
    );
  }

// Created this function to allow the openDrawer function to be called from the header in MainScreen.js
function openDrawer() {
    _navigator.dispatch(DrawerActions.openDrawer())
}

export default {
  navigate,
  navigateDrawer,
  setTopLevelNavigator,
  getTopLevelNavigator,
  openDrawer,
};