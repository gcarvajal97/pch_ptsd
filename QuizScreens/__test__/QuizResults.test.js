import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import renderer from "react-test-renderer";
import NavigationService from "../../components/NavigationService";
import QuizResults from "../QuizResults";
import translate from "../../components/translateService";

jest.mock("../../components/NavigationService.js");
jest.mock("../../components/translateService.js");

describe("QuizResults", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("renders correctly", () => {
        let navigation = {
            getParam: (param, value) => {
                return {};
            }
        };
        expect(
            renderer.create(<QuizResults navigation={navigation} />).toJSON()
        ).toMatchSnapshot();
    });

    it("displays a message if no quiz questions were answered", () => {
        let navigation = {
            getParam: (param, value) => {
                return {};
            }
        };
        renderer.create(<QuizResults navigation={navigation} />);
        expect(translate).toHaveBeenCalledWith("quizResults.noAnswers");
    });

    it("displays cards if quiz responses are valid", () => {
        let navigation = {
            getParam: (param, value) => {
                return {
                    1: 0,
                    2: 1,
                };
            }
        };
        renderer.create(<QuizResults navigation={navigation} />);
        expect(translate).not.toHaveBeenCalledWith("quizResults.noAnswers");
        expect(translate).toHaveBeenCalledWith("quizResults.q1title")
        expect(translate).toHaveBeenCalledWith("quizResults.q2title")
        expect(translate).toHaveBeenCalledWith("quizResults.q1body1")
        expect(translate).toHaveBeenCalledWith("quizResults.q2body1")
        expect(translate).toHaveBeenCalledWith("quizResults.q1body2")
        expect(translate).toHaveBeenCalledWith("quizResults.q2body2")
        expect(translate).toHaveBeenCalledWith("quizResults.whyHappens")
        expect(translate).toHaveBeenCalledWith("quizResults.isProblem")
    });

    it("do nothing if quiz responses are invalid", () => {
        let navigation = {
            getParam: (param, value) => {
                return {
                    1: -1,
                    2: 2,
                };
            }
        };
        renderer.create(<QuizResults navigation={navigation} />);
        expect(translate).not.toHaveBeenCalled();
    });
});
