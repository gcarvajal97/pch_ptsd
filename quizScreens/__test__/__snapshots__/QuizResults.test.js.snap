// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizResults renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "elevation": 4,
      "flex": 1,
      "margin": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        accessible={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "avenir-heavy",
                "fontSize": 14,
                "marginTop": 5,
                "textAlign": "left",
              }
            }
          >
             
             
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 14,
                "marginTop": 5,
                "paddingHorizontal": 5,
                "paddingVertical": 1,
              }
            }
          >
             
             
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "avenir-heavy",
                "fontSize": 14,
                "marginTop": 5,
                "textAlign": "left",
              }
            }
          >
             
             
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 14,
                "marginTop": 5,
                "paddingHorizontal": 5,
                "paddingVertical": 1,
              }
            }
          >
             
             
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
